### PR TITLE
Updates to Custom Types Chapter

### LINKED ISSUE #70: Add a section on constructors to the User Defined Types chapter
The exercise asks students to create a constructor, but the chapter doesn't mention them.

### PR DESCRIPTION
**Pull request by askingalot**
Added sections for constructors and fields.
Updated description of types.
Updated Writer class example.

Resolves #70

### CODE REVIEW

File: book-1-orientation/chapters/CLASSES_INTRO.md [hunk 1]
```diff
@@ -1,27 +1,42 @@
...
+Writer author = new Writer("I do declare: ");
+author.Write(output);
+
+author.Prefix = "That's right, I said: ";
+author.Write(output);
 ```
+The above code would print the following:
+
+    I do declare: Nashville Software School
+    That's right, I said:: Nashville Software School
```
Comment (copypastedeveloper) [inline book-1-orientation/chapters/CLASSES_INTRO.md]: super pedantic, but there shouldn't be 2 colons printed.

File: book-1-orientation/chapters/CLASSES_INTRO.md [hunk 2]
```diff
@@ -281,6 +298,62 @@ namespace Classes
 }
 ```
 
+## Fields
+
+Much like properties, Fields are use to store values in an object. Unlike properties fields are usually private and cannot be accessed outside the object. The concept of storing private data that is only accessible inside the object is referred to as "encapsulation".
```
Comment (copypastedeveloper) [inline book-1-orientation/chapters/CLASSES_INTRO.md]: Much like `properties`, `fields` are used to store values in an object.  Whereas properties are generally used to expose public state, fields are usually private and are used to store internal, or private, state.  This concept of storing private data that can only be accessed from inside the object is known as `encapsulation`.

Comment (stevebrownlee) [reply book-1-orientation/chapters/CLASSES_INTRO.md]: 👍

File: book-1-orientation/chapters/CLASSES_INTRO.md (hunk 1 above)
Comment (stevebrownlee) [reply book-1-orientation/chapters/CLASSES_INTRO.md]: 👍

**Comment by stevebrownlee** [comment]
This is a good next version of the chapter. I made one suggestion, and Nate made several. Incorporate those and merge it.

File: book-1-orientation/chapters/CLASSES_INTRO.md (hunk 1 above)
Comment (askingalot) [reply book-1-orientation/chapters/CLASSES_INTRO.md]: Thanks for the proofreading.

Unless I missed something I don't think access modifiers have been mentioned prior to this chapter. Personally, I feel like that part of this chapter needs some work, having never taught it, I don't have ideas as to what needs to be changed. I decided to focus on the changes I was more certain about.

File: book-1-orientation/chapters/CLASSES_INTRO.md (hunk 2 above)
Comment (askingalot) [reply book-1-orientation/chapters/CLASSES_INTRO.md]: I'd rather make those words **bold** instead. I think using the `code` highlight suggests that those are keywords.

**Comment by askingalot** [comment]
Thanks, Steve.

File: book-1-orientation/chapters/CLASSES_INTRO.md (hunk 2 above)
Comment (copypastedeveloper) [reply book-1-orientation/chapters/CLASSES_INTRO.md]: that makes a lot of sense.  I'm for bold.  I just use ticks as my go to text differentiator cause i'm lazy.

**Pull request merged**